### DISCUSSION TITLE
Viewing the designed structure with Python

### DISCUSSION THREAD

**Discussion by Elithraldor** [thliebig/openEMS-Project#118] [Q&A Usage] [answered]
I must admit it is sometime since I last used openEMS, and at that time it was with Octave. Now, I seem to remember that it was possible to view a 3D model of your structure - I find this very useful to check whether I made some kind of mistake in designing my antennas.

It seems that AppCSXCAD is able to view it, but it is only for Linux? I am on Windows 10/11.

Is it is possible to view the 3D structure using Windows and Python? Or should I use Octave instead?

I should perhaps note that I have the python tutorials working just fine.

Thank you for any help.

**Comment by biergaizi** [comment, marked as answer]
It should definitely be possible. If it does not, then it's a bug and needs troubleshooting.

**Reply by Elithraldor** [reply]
It's just me who's blind as a bat, in the Simple_Patch_antenna.py tutorial I just needed to "enable" debugging:
```
### Run the simulation
if 1:  # debugging only
    CSX_file = os.path.join(Sim_Path, 'simp_patch.xml')
    if not os.path.exists(Sim_Path):
        os.mkdir(Sim_Path)
    CSX.Write2XML(CSX_file)
    from CSXCAD import AppCSXCAD_BIN
    os.system(AppCSXCAD_BIN + ' "{}"'.format(CSX_file))
```

And the AppCSXCAD is part of the zip file. So no problems at all.
